notebooks/demo_latticemaze.ipynb

```diff
@@ -211,17 +225,20 @@
             "outputs": [],
             "source": [
                 "MazePlot(maze).plot()\n",
-                "plt.show()"
+                "plt.show()\n",
+                "\n",
+                "plt.imshow(maze.as_pixels())\n",
+                "plt.show()\n",
+                "\n",
+                "print(maze.as_ascii())"
             ]
```

Comment:
High level question: Where do we want the plotting functionality to live? As part of LatticeMaze or and external plotting class (MazePlot)?

As Michael commented, we currently have redundancy for 
- LatticeMaze.to_pixels and MazePlot.plot()
- LatticeMaze.as_ascii and MazePlot.as_ascii

Discussion:
I agree we dont want to have much overhead if one just wants to display a sample maze on the fly (overhead being creating a separate instance of MazePlot class for instance).
However, we discussed earlier, separating plotting functionality to a separate file than LatticeMaze would allow for better testing and more organized/clear code. What are people's thoughts on this?

Reply:
part of my reason for including the `{as/from}_pixels` and `{as/from}_ascii` in the actual underlying `LatticeMaze` and derived classes is that the pixel and ascii representations might be useful not just for human viewing but also for training models and creating datasets. Notably:
- @afspies needs the `from_ascii` or `from_pixels` so that we can create test mazes from his pygame app -- it would be unwieldy to have this in the mazeplot, which then returns a modified `LatticeMaze` or `TargetedLatticeMaze`
- training a model on a rasterized representation of the maze, like in `to_pixels()`, was briefly discussed as something we might want to do
- a different project I am involved in (solving mazes via an architecture derived from RCNNs with some nicer mathematical properties) will probably make use of the rasterized representation

Reply:
That makes sense. In that case, we keep the redundancy of the functions below?

- LatticeMaze.as_pixels and MazePlot.plot()
- LatticeMaze.as_ascii and MazePlot.as_ascii

Reply:
I think we should definitely keep that as part of the interface to `MazePlot`, but perhaps to minimize code duplication have those functions call the underlying functions for `LatticeMaze` if possible.